Sources/MapboxSearch/PublicAPI/Use Cases/Address Autofill/AddressAutofill.swift

```diff
@@ -105,13 +105,16 @@ public extension AddressAutofill {
     ) {
         userActivityReporter.reportActivity(forComponent: "address-autofill-suggestion-select")
 
-        let result = AddressAutofill.Result(
-            name: suggestion.name,
-            formattedAddress: suggestion.formattedAddress,
-            coordinate: suggestion.coordinate,
-            addressComponents: suggestion.addressComponents
-        )
-        completion(.success(result))
+        guard case let .suggestion(coreSearch, coreOptions) = suggestion.underlying else {
```

Comment:
What about the case when `underlying` is `.result` case?  Are such cases invalid? Should then `.result` enum case be removed?

Reply:
- For AddressAutofill the `Underlying.result` enum case is still used by the `fetchSuggestions(using: completion:)` function which calls the `reverseGeocoding` API. 
- Similarly for PlaceAutocomplete the result is still necessary to use the `select(suggestion: completion:)` function.

Reply:
I have an idea for some more improvements in this function later today or tomorrow